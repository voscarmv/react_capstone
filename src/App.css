.width-auto {
  width: 800px !important;
}
