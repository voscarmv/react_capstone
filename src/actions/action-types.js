export const FETCH_BREEDS_REQUEST = 'FETCH_BREEDS_REQUEST';
export const FETCH_BREEDS_SUCCESS = 'FETCH_BREEDS_SUCCESS';
export const FETCH_BREEDS_ERROR = 'FETCH_BREEDS_ERROR';
export const FETCH_CAT_REQUEST = 'FETCH_CAT_REQUEST';
export const FETCH_CAT_SUCCESS = 'FETCH_CAT_SUCCESS';
export const FETCH_CAT_ERROR = 'FETCH_CAT_ERROR';
export const UPDATE_FILTER = 'UPDATE_FILTER';
