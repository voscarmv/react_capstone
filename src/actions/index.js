import fetch from 'node-fetch';
import {
  FETCH_BREEDS_REQUEST,
  FETCH_BREEDS_SUCCESS,
  FETCH_BREEDS_ERROR,
  FETCH_CAT_REQUEST,
  FETCH_CAT_SUCCESS,
  FETCH_CAT_ERROR,
  UPDATE_FILTER,
} from './action-types';

export const updateFilter = payload => ({ type: UPDATE_FILTER, payload });
export const fetchBreedsRequest = () => ({ type: FETCH_BREEDS_REQUEST });
export const fetchBreedsSuccess = breeds => ({ type: FETCH_BREEDS_SUCCESS, payload: breeds });
export const fetchBreedsError = error => ({ type: FETCH_BREEDS_ERROR, payload: error });
export const fetchCatRequest = () => ({ type: FETCH_CAT_REQUEST });
export const fetchCatSuccess = cat => ({ type: FETCH_CAT_SUCCESS, payload: cat });
export const fetchCatError = error => ({ type: FETCH_CAT_ERROR, payload: error });
export const fetchBreeds = () => async dispatch => {
  dispatch(fetchBreedsRequest());
  try {
    const getBreeds = await fetch('https://api.thecatapi.com/v1/breeds');
    const breedsJSON = await getBreeds.json();
    dispatch(fetchBreedsSuccess(breedsJSON));
  } catch (e) {
    dispatch(fetchBreedsError(e));
  }
};
export const fetchCat = breed => async dispatch => {
  dispatch(fetchCatRequest());
  const xAPIKey = 'x-api-key';
  try {
    const getCat = await fetch(
      `https://api.thecatapi.com/v1/images/search?breed_id=${breed}`,
      {
        headers: {
          [xAPIKey]: '20874f76-4233-4e78-9f44-e30c018c07e8',
        },
      },
    );
    const catJSON = await getCat.json();
    dispatch(fetchCatSuccess(catJSON));
  } catch (e) {
    dispatch(fetchCatError(e));
  }
};
