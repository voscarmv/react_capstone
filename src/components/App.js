import React from 'react';
import List from './List';
import Form from './Form';
import BreedMenu from './BreedMenu';
import BreedFilter from './BreedFilter';

const App = () => (
  <div>
    <List />
    <Form />
    <BreedFilter />
    <BreedMenu />
  </div>
);

export default App;
