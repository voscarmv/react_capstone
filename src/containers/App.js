import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Nav from './Nav';
import Cat from './Cat';
import Home from './Home';

const App = () => (
  <Router>
    <Nav />
    <Switch>
      <Route exact path="/">
        <Home />
      </Route>
      <Route path="/cat">
        <Cat />
      </Route>
    </Switch>
  </Router>
);

export default App;
