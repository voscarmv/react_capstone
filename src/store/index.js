import { createStore, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import combineReducers from '../reducers/index';

const store = createStore(
  combineReducers,
  composeWithDevTools(
    applyMiddleware(
      logger,
      thunk,
    ),
  ),
);

export default store;
