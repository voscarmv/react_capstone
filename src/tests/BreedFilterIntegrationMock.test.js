import React from 'react';
import {
  render, screen, waitFor, fireEvent,
} from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../store/index';
import App from '../containers/App';
import 'regenerator-runtime/runtime';
import * as actions from '../actions/index';
import breeds from './BreedsResponse';
// import cat from './CatResponse';

jest.mock('../actions/index');

describe('App', () => {
  beforeEach(() => {
    actions.fetchBreeds.mockImplementation(
      () => ({ type: 'FETCH_BREEDS_SUCCESS', payload: breeds }),
    );
    actions.updateFilter.mockImplementation(
      () => ({ type: 'UPDATE_FILTER', payload: { property: 'intelligence', value: '1' } }),
    );
  });
  afterEach(() => {
    actions.fetchBreeds.mockRestore();
    actions.updateFilter.mockRestore();
  });
  test('renders filter component', async () => {
    const fullApp = render(<Provider store={store}><App /></Provider>);
    const { getByTestId, asFragment } = fullApp;
    fireEvent.click(getByTestId('breed_selector'));
    await waitFor(
      () => {
        expect(screen.getByText('Siberian'));
      },
    );
    fireEvent.click(getByTestId('intelligence_enable'));
    await waitFor(
      () => {
        expect(screen.getByTestId('intelligence').classList.contains('d-none')).toBe(false);
      },
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
