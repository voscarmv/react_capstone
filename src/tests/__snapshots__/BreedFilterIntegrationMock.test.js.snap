// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders filter component 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand navbar-dark bg-dark"
  >
    <span
      class="navbar-brand"
    >
      <span
        aria-label="cat"
        role="img"
      >
        😺
      </span>
      <a
        href="/"
      >
        The Cat-alog
      </a>
    </span>
    <div
      class="container-fluid"
    >
      <div
        class="ml-auto text-light"
      >
        Show breed:
      </div>
      <div
        class="p-3"
      >
        <div
          class="dropdown"
        >
          <button
            aria-expanded="false"
            aria-haspopup="true"
            class="dropdown-toggle btn btn-success"
            data-testid="breed_selector"
            id="dropdown-basic"
            type="button"
          >
            Choose
          </button>
          <div
            aria-labelledby="dropdown-basic"
            class="dropdown-menu"
            style="margin: 0px;"
          />
        </div>
      </div>
      <div
        class=""
      >
        <button
          aria-controls="collapse-filter-form"
          aria-expanded="false"
          class="btn btn-primary"
          type="button"
        >
          Filter
        </button>
      </div>
    </div>
  </nav>
  <div
    class="collapse"
    id="collapse-filter-form"
  >
    <div
      class="bg-dark text-light p-3"
    >
      <form>
        <div
          class="ml-auto mr-3 font-weight-bold"
        >
          Filter by
        </div>
        <label
          class="ml-auto"
          for="experimental"
        >
          Experimental
          <input
            class="mr-3 ml-1"
            data-testid="experimental_enable"
            id="experimental"
            type="checkbox"
            value="experimental"
          />
        </label>
        <label
          class="ml-auto"
          for="hairless"
        >
          Hairless
          <input
            class="mr-3 ml-1"
            data-testid="hairless_enable"
            id="hairless"
            type="checkbox"
            value="hairless"
          />
        </label>
        <label
          class="ml-auto"
          for="natural"
        >
          Natural
          <input
            class="mr-3 ml-1"
            data-testid="natural_enable"
            id="natural"
            type="checkbox"
            value="natural"
          />
        </label>
        <label
          class="ml-auto"
          for="rare"
        >
          Rare
          <input
            class="mr-3 ml-1"
            data-testid="rare_enable"
            id="rare"
            type="checkbox"
            value="rare"
          />
        </label>
        <label
          class="ml-auto"
          for="suppressed_tail"
        >
          Suppressed tail
          <input
            class="mr-3 ml-1"
            data-testid="suppressed_tail_enable"
            id="suppressed_tail"
            type="checkbox"
            value="suppressed_tail"
          />
        </label>
        <label
          class="ml-auto"
          for="short_legs"
        >
          Short legs
          <input
            class="mr-3 ml-1"
            data-testid="short_legs_enable"
            id="short_legs"
            type="checkbox"
            value="short_legs"
          />
        </label>
        <label
          class="ml-auto"
          for="hypoallergenic"
        >
          Hypoallergenic
          <input
            class="mr-3 ml-1"
            data-testid="hypoallergenic_enable"
            id="hypoallergenic"
            type="checkbox"
            value="hypoallergenic"
          />
        </label>
        <label
          class="ml-auto"
          for="adaptability"
        >
          Adaptability
          <input
            class="mr-3 ml-1"
            data-testid="adaptability_enable"
            id="adaptability"
            type="checkbox"
            value="adaptability"
          />
        </label>
        <label
          class="ml-auto"
          for="affection_level"
        >
          Affection level
          <input
            class="mr-3 ml-1"
            data-testid="affection_level_enable"
            id="affection_level"
            type="checkbox"
            value="affection_level"
          />
        </label>
        <label
          class="ml-auto"
          for="child_friendly"
        >
          Child friendly
          <input
            class="mr-3 ml-1"
            data-testid="child_friendly_enable"
            id="child_friendly"
            type="checkbox"
            value="child_friendly"
          />
        </label>
        <label
          class="ml-auto"
          for="dog_friendly"
        >
          Dog friendly
          <input
            class="mr-3 ml-1"
            data-testid="dog_friendly_enable"
            id="dog_friendly"
            type="checkbox"
            value="dog_friendly"
          />
        </label>
        <label
          class="ml-auto"
          for="energy_level"
        >
          Energy level
          <input
            class="mr-3 ml-1"
            data-testid="energy_level_enable"
            id="energy_level"
            type="checkbox"
            value="energy_level"
          />
        </label>
        <label
          class="ml-auto"
          for="grooming"
        >
          Grooming
          <input
            class="mr-3 ml-1"
            data-testid="grooming_enable"
            id="grooming"
            type="checkbox"
            value="grooming"
          />
        </label>
        <label
          class="ml-auto"
          for="health_issues"
        >
          Health issues
          <input
            class="mr-3 ml-1"
            data-testid="health_issues_enable"
            id="health_issues"
            type="checkbox"
            value="health_issues"
          />
        </label>
        <label
          class="ml-auto"
          for="intelligence"
        >
          Intelligence
          <input
            class="mr-3 ml-1"
            data-testid="intelligence_enable"
            id="intelligence"
            type="checkbox"
            value="intelligence"
          />
        </label>
        <label
          class="ml-auto"
          for="shedding_level"
        >
          Shedding level
          <input
            class="mr-3 ml-1"
            data-testid="shedding_level_enable"
            id="shedding_level"
            type="checkbox"
            value="shedding_level"
          />
        </label>
        <label
          class="ml-auto"
          for="social_needs"
        >
          Social needs
          <input
            class="mr-3 ml-1"
            data-testid="social_needs_enable"
            id="social_needs"
            type="checkbox"
            value="social_needs"
          />
        </label>
        <label
          class="ml-auto"
          for="stranger_friendly"
        >
          Stranger friendly
          <input
            class="mr-3 ml-1"
            data-testid="stranger_friendly_enable"
            id="stranger_friendly"
            type="checkbox"
            value="stranger_friendly"
          />
        </label>
        <label
          class="ml-auto"
          for="vocalisation"
        >
          Vocalisation
          <input
            class="mr-3 ml-1"
            data-testid="vocalisation_enable"
            id="vocalisation"
            type="checkbox"
            value="vocalisation"
          />
        </label>
        <hr />
        <div
          class="ml-auto mr-3 font-weight-bold"
        >
          Filter preferences
        </div>
        <label
          class="ml-auto d-none"
          data-testid="experimental"
          for="experimental"
        >
          <span>
            Experimental
          </span>
          <input
            class="mr-3 ml-1"
            disabled=""
            id="experimental"
            type="checkbox"
            value="experimental"
          />
        </label>
        <label
          class="ml-auto d-none"
          data-testid="hairless"
          for="hairless"
        >
          <span>
            Hairless
          </span>
          <input
            class="mr-3 ml-1"
            disabled=""
            id="hairless"
            type="checkbox"
            value="hairless"
          />
        </label>
        <label
          class="ml-auto d-none"
          data-testid="natural"
          for="natural"
        >
          <span>
            Natural
          </span>
          <input
            class="mr-3 ml-1"
            disabled=""
            id="natural"
            type="checkbox"
            value="natural"
          />
        </label>
        <label
          class="ml-auto d-none"
          data-testid="rare"
          for="rare"
        >
          <span>
            Rare
          </span>
          <input
            class="mr-3 ml-1"
            disabled=""
            id="rare"
            type="checkbox"
            value="rare"
          />
        </label>
        <label
          class="ml-auto d-none"
          data-testid="suppressed_tail"
          for="suppressed_tail"
        >
          <span>
            Suppressed tail
          </span>
          <input
            class="mr-3 ml-1"
            disabled=""
            id="suppressed_tail"
            type="checkbox"
            value="suppressed_tail"
          />
        </label>
        <label
          class="ml-auto d-none"
          data-testid="short_legs"
          for="short_legs"
        >
          <span>
            Short legs
          </span>
          <input
            class="mr-3 ml-1"
            disabled=""
            id="short_legs"
            type="checkbox"
            value="short_legs"
          />
        </label>
        <label
          class="ml-auto d-none"
          data-testid="hypoallergenic"
          for="hypoallergenic"
        >
          <span>
            Hypoallergenic
          </span>
          <input
            class="mr-3 ml-1"
            disabled=""
            id="hypoallergenic"
            type="checkbox"
            value="hypoallergenic"
          />
        </label>
        <label
          class="ml-auto d-none"
          data-testid="adaptability"
          for="adaptability"
        >
          <span>
            Adaptability
          </span>
          <input
            class="mr-3 ml-1"
            disabled=""
            id="adaptability"
            max="5"
            min="1"
            type="number"
            value="1"
          />
        </label>
        <label
          class="ml-auto d-none"
          data-testid="affection_level"
          for="affection_level"
        >
          <span>
            Affection level
          </span>
          <input
            class="mr-3 ml-1"
            disabled=""
            id="affection_level"
            max="5"
            min="1"
            type="number"
            value="1"
          />
        </label>
        <label
          class="ml-auto d-none"
          data-testid="child_friendly"
          for="child_friendly"
        >
          <span>
            Child friendly
          </span>
          <input
            class="mr-3 ml-1"
            disabled=""
            id="child_friendly"
            max="5"
            min="1"
            type="number"
            value="1"
          />
        </label>
        <label
          class="ml-auto d-none"
          data-testid="dog_friendly"
          for="dog_friendly"
        >
          <span>
            Dog friendly
          </span>
          <input
            class="mr-3 ml-1"
            disabled=""
            id="dog_friendly"
            max="5"
            min="1"
            type="number"
            value="1"
          />
        </label>
        <label
          class="ml-auto d-none"
          data-testid="energy_level"
          for="energy_level"
        >
          <span>
            Energy level
          </span>
          <input
            class="mr-3 ml-1"
            disabled=""
            id="energy_level"
            max="5"
            min="1"
            type="number"
            value="1"
          />
        </label>
        <label
          class="ml-auto d-none"
          data-testid="grooming"
          for="grooming"
        >
          <span>
            Grooming
          </span>
          <input
            class="mr-3 ml-1"
            disabled=""
            id="grooming"
            max="5"
            min="1"
            type="number"
            value="1"
          />
        </label>
        <label
          class="ml-auto d-none"
          data-testid="health_issues"
          for="health_issues"
        >
          <span>
            Health issues
          </span>
          <input
            class="mr-3 ml-1"
            disabled=""
            id="health_issues"
            max="5"
            min="1"
            type="number"
            value="1"
          />
        </label>
        <label
          class="ml-auto"
          data-testid="intelligence"
          for="intelligence"
        >
          <span>
            Intelligence
          </span>
          <input
            class="mr-3 ml-1"
            id="intelligence"
            max="5"
            min="1"
            type="number"
            value="1"
          />
        </label>
        <label
          class="ml-auto d-none"
          data-testid="shedding_level"
          for="shedding_level"
        >
          <span>
            Shedding level
          </span>
          <input
            class="mr-3 ml-1"
            disabled=""
            id="shedding_level"
            max="5"
            min="1"
            type="number"
            value="1"
          />
        </label>
        <label
          class="ml-auto d-none"
          data-testid="social_needs"
          for="social_needs"
        >
          <span>
            Social needs
          </span>
          <input
            class="mr-3 ml-1"
            disabled=""
            id="social_needs"
            max="5"
            min="1"
            type="number"
            value="1"
          />
        </label>
        <label
          class="ml-auto d-none"
          data-testid="stranger_friendly"
          for="stranger_friendly"
        >
          <span>
            Stranger friendly
          </span>
          <input
            class="mr-3 ml-1"
            disabled=""
            id="stranger_friendly"
            max="5"
            min="1"
            type="number"
            value="1"
          />
        </label>
        <label
          class="ml-auto d-none"
          data-testid="vocalisation"
          for="vocalisation"
        >
          <span>
            Vocalisation
          </span>
          <input
            class="mr-3 ml-1"
            disabled=""
            id="vocalisation"
            max="5"
            min="1"
            type="number"
            value="1"
          />
        </label>
      </form>
    </div>
  </div>
  <div
    class="m-3"
  >
    <h5>
      Welcome to The Cat-alog!
      <span
        aria-label="cat"
        role="img"
      >
        🐈
      </span>
    </h5>
    <p>
      You can browse through our database of cat breeds by using the filter on the top-right corner of this page.
    </p>
    <p>
      Simply select a breed from the drop-down menu to display its details here. Or if you prefer, filter them first by clicking on the Filter button and setting the parameters you are looking for!
    </p>
  </div>
</DocumentFragment>
`;
